micropython/lora/lora/lora/modem.py

```diff
@@ -0,0 +1,537 @@
+            if timeout_ms <= 0:
+                # Timed out in software, nothing to resume
+                self._end_receive()
+                if _DEBUG:
+                    print("Timed out in software timeout_ms={}".format(timeout_ms))
+                schedule(
+                    self._radio_isr, None
+                )  # "soft irq" to unblock anything waiting on the interrupt event
+                return False
+
+        if _DEBUG:
+            print(
+                "Resuming receive timeout_ms={} continuous={} rx_length={}".format(
+                    timeout_ms, self._rx_continuous, self._rx_length
+                )
+            )
+
+        self.start_receive(timeout_ms, self._rx_continuous, self._rx_length)
+
+        # restore the previous version of _rx so ticks_ms deadline can't
+        # slowly creep forward each time this happens
+        self._rx = rx
+
+        return True
+
+    # Common parts of transmit API
+
+    def poll_transmit(self):
+        # Check the ongoing transmit state.
+        #
+        # Returns one of:
+        #
+        # - True if a transmit is ongoing and the caller
+        #   should call again.
+        # - False if no transmit is ongoing.
+        # - An int value exactly one time per transmission, the first time
+        #   poll_transmit() is called after a transmit ends. In this case it
+        #   is the utime.ticks_ms() timestamp of the time that the transmit completed.
+        #
+        # Note this function only returns an int value one time (the first time it
+        # is called after transmit completes).
+        #
+        # Part of common low-level modem API, see README.md for usage.
+        if not self._tx:
+            return False
+
+        ticks_ms = self._get_last_irq()
+
+        if not (self._get_irq() & self._IRQ_TX_COMPLETE):
+            # Not done. If the host and modem get out
+            # of sync here, or the caller doesn't follow the sequence of
+            # transmit operations exactly, then can end up in a situation here
+            # where the modem has stopped transmitting and has gone to Standby,
+            # so _IRQ_TX_DONE is never set.
+            #
+            # For now, leaving this for the caller to do correctly. But if it becomes an issue then
+            # we can call _get_mode() here as well and check the modem is still in a TX mode.
+            return True
+
+        self._clear_irq()
+
+        self._tx = False
+
+        if self._ant_sw:
+            self._ant_sw.idle()
+
+        # The modem just finished transmitting, so start receiving again if needed
+        self._check_receive()
+
+        return ticks_ms
+
+    # Simple synchronous modem API functions
+    #
+    # These are intended for simple applications. They block the caller until
+    # the modem operation is complete, and don't support interleaving transmit
+    # and receive.
+
+    def transmit(self, packet, tx_at_ms=None):
```

Comment:
IMO (and @jimmo agrees), this driver (and all its components) should use `send` and `recv` naming, like a `socket.socket` does.  (It's not a stream so not read/write.)

That would mean this method is called `send`, and the method below `recv`.

Then it would also have `prepare_send`, `start_send`, `poll_send`, `start_recv`, `poll_recv`, `recv_continuous` etc.

Reply:
Some of the reasons for this are:
1. more Pythonic, matches naming elsewhere
2. reduces code size because the names already exist as qstrs